src/persist-client/src/internal/maintenance.rs

```diff
@@ -49,6 +50,13 @@ pub struct RoutineMaintenance {
 }
 
 impl RoutineMaintenance {
+    pub(crate) fn is_empty(&self) -> bool {
```

Comment:
I worry about this rotting. one option is to bring back the PartialEq and instead of this method, do `== RoutineMaintenance::default()`. another is to destructure it, so we at least get a compile error if a new field is added (the latter is a pretty common pattern in persist code)

```
// Destructure so that...
let RoutineMaintenance{garbage_collection, lease_expiration: LeaseExpiration{readers: lease_expiration_readers, writers: lease_expiration_writers}, write_rollup} == self;
garbage_collection.is_none() && lease_expiration_readers.is_empty() && ...
```

I actually lean a bit toward the former

Reply:
I don't have strong feelings; happy to go with your gut on this one.